00_ES6/00_start/src/pages/lobby/lobby.container.jsx

```diff
@@ -1,41 +1,69 @@
 import React from 'react';
-import { getListOfRooms } from '../../api/rooms';
+import { withRouter } from 'react-router-dom';
+import { SessionContext } from '../../common';
+import { getListOfRooms, canenrollRoom } from '../../api/rooms';
 import { LobbyComponent } from './lobby.component';
 
-export class LobbyContainer extends React.Component {
+class LobbyContainerInner extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {rooms: [], nickname: '', selectedRoom: {id: 0, name:''}};
+    this.state = { rooms: [], nickname: '', selectedRoom: '' };
 
     this.fetchRooms = this.fetchRooms.bind(this);
   }
 
   async fetchRooms() {
     const rooms = await getListOfRooms();
-    this.setState({rooms});
+    this.setState({ rooms });
+  }
+
+  async joinRoomRequest() {
+    const canEnroll = await canenrollRoom(this.state.selectedRoom, this.state.nickname);
+
+    if (canEnroll) {
+      console.log(`*** Join Room Request succeeded
+      Nickname: ${this.state.nickname}
+      Room: ${this.state.selectedRoom}`);
+
+      this.props.setChatSessionInfo(this.state.nickname, this.state.selectedRoom);
+      this.props.history.push('/chat');
+
+    } else {
+      console.log(`Join room request failed try another nickname`);
+    }
   }
 
   onFieldChange = (id) => (value) => {
-    this.setState({[id]: value})
+    this.setState({ [id]: value })
   }
 
   onJoinRoomRequest = () => {
-    console.log(`*** Join Room Request 
-                 Nickname: ${this.state.nickname}
-                 Room: ${this.state.selectedRoom.name}`);
+    this.joinRoomRequest();
   }
 
   render() {
     return (
-      <LobbyComponent 
-        rooms={this.state.rooms} 
+      <LobbyComponent
+        rooms={this.state.rooms}
         fetchRooms={this.fetchRooms}
         nickname={this.state.nickname}
         onFieldChange={this.onFieldChange}
         selectedRoom={this.state.selectedRoom}
         onJoinRoomRequest={this.onJoinRoomRequest}
-        />
+      />
     );
   }
 }
 
+const LobbyContainerWithSessionInner = (props) =>
+      <SessionContext.Consumer>
+       {(sessionInfo) => <LobbyContainerInner 
+                          setChatSessionInfo={sessionInfo.setChatSessionInfo}
+                          {...props}/>
+       }
+      </SessionContext.Consumer>
+    
+
```

Comment:
I thnk I should add here prop types, lobby container should recieve setChatSession and history

Once migrated to redux it will recieve only props.history? or should we go for reat-router-redux?

What do you think?